src/MagicChunks.Tests/Documents/YamlDocumentTests.cs

```diff
@@ -111,26 +111,40 @@ public void Remove()
         public void ValidateEmptyPath()
         {
             // Assert
-            YamlDocument document = new YamlDocument("");
+            var document = new YamlDocument(@"a:
+    x: 1
+b: 2
+c: 3");
```

Comment:
I moved the check for the lack of a root object from the call to ReplaceKey to the constructor since it could be checked with no other data required.

Because of this, the constructor now throws the Root Element Not Present error.

This test was modified to only validate the empty path exception, and a new test was added below to verify that the no root element exception is thrown by the constructor.